### PR TITLE
FEI-5298: Replace Coveralls with Codecov

### PR DESCRIPTION
**Pull request by kevinbarabash**
## Summary:
We've found Coveralls to be too flaky to be useful.  It often hangs and sometimes it reports drops in coverages in types-only changes.

This PR switches back to Codecov but only uploads a single report for jest.  In the future we'll also generate coverage for cypress and merge the reports ourselves.  This PR is just to get a report of some sort showing up in the PRs.

Issue: FEI-5298

## Test plan:
- make a trivial change to a source code file, see that the report is uploaded to Codecov

### CODE REVIEW

File: packages/perseus/src/widgets/__tests__/expression.test.tsx
Comment (jeremywiebe) [inline packages/perseus/src/widgets/__tests__/expression.test.tsx:183]: Can we bring this back or is it flaky/broken in some way?

File: .prettierignore
Comment (jeremywiebe) [inline .prettierignore:33]: I don't see this path used anywhere in this PR. Does the codecov-action just write to here by default?

**Review by jeremywiebe — APPROVED** [review]
Thanks for flipping this back. Hopefully it's less hangy.

File: packages/perseus/src/widgets/__tests__/expression.test.tsx
Comment (kevinbarabash) [reply packages/perseus/src/widgets/__tests__/expression.test.tsx:183]: I'll renable this before landing.

File: .prettierignore
Comment (kevinbarabash) [reply .prettierignore:33]: It's possible it's something that's only on my machine because I was playing around with some other tooling.  I get rid of it and delete the dir locally.

File: .github/workflows/node-ci-main.yml
Comment (kevinbarabash) [inline .github/workflows/node-ci-main.yml:57]: @jeremywiebe should I set `continue-on-error: true` for this step?  I know some people had concerns about coverage drops show up as red `x`s on PRs.  Personally, I think we should be using required checks for those checks that we think are important enough to block merging.

Comment (jeremywiebe) [reply .github/workflows/node-ci-main.yml:57]: 🤔  I don't think we should set `continue-on-error` to begin with. I'd rather tune the `target` value in our `codecov.yml` file if it's too sensitive.  

@handeyeco We can discuss and [disable this check](https://docs.codecov.com/docs/commit-status#informational) but still get the information even after Kevin lands this PR.

Comment (kevinbarabash) [reply .github/workflows/node-ci-main.yml:57]: Things should fall in the green zone with the current settings once the cypress coverage results are merged in.

**Pull request merged**
